tests/unit/cypher_main_visitor.cpp

```diff
@@ -1032,18 +1032,52 @@ TEST_P(CypherMainVisitorTest, MapLiteral) {
 TEST_P(CypherMainVisitorTest, MapProjectionLiteral) {
   auto &ast_generator = *GetParam();
   auto *query = dynamic_cast<CypherQuery *>(ast_generator.ParseQuery(
-      "WITH {name: \"Morgan\"} as actor, 85 as age RETURN actor {.name, lastName: \"Freeman\", age}"));
+      "WITH {name: \"Morgan\"} as actor, 85 as age RETURN actor {.name, .*, age, lastName: \"Freeman\"}"));
   ASSERT_TRUE(query);
   ASSERT_TRUE(query->single_query_);
   auto *single_query = query->single_query_;
   auto *return_clause = dynamic_cast<Return *>(single_query->clauses_[1]);
   auto *map_projection_literal =
       dynamic_cast<MapProjectionLiteral *>(return_clause->body_.named_expressions[0]->expression_);
   ASSERT_TRUE(map_projection_literal);
-  ASSERT_EQ(3, map_projection_literal->elements_.size());
-  // ast_generator.CheckLiteral(map_projection_literal->elements_[ast_generator.Prop("name")], "Morgan");
-  // ast_generator.CheckLiteral(map_projection_literal->elements_[ast_generator.Prop("lastName")], "Freeman");
-  // ast_generator.CheckLiteral(map_projection_literal->elements_[ast_generator.Prop("age")], "85");
+  ASSERT_EQ(4, map_projection_literal->elements_.size());
+  ASSERT_EQ(map_projection_literal->elements_[ast_generator.Prop("name")]->GetTypeInfo().name, "PropertyLookup");
+  ASSERT_EQ(map_projection_literal->elements_[ast_generator.Prop("*")]->GetTypeInfo().name, "AllPropertiesLookup");
+  ASSERT_EQ(map_projection_literal->elements_[ast_generator.Prop("age")]->GetTypeInfo().name, "Identifier");
+  ASSERT_EQ(map_projection_literal->elements_[ast_generator.Prop("lastName")]->GetTypeInfo().name,
+            std::string(typeid(ast_generator).name()).ends_with("CachedAstGenerator") ? "ParameterLookup"
+                                                                                      : "PrimitiveLiteral");
```

Comment:
NB for reviewers: `CachedAstGenerator` turns `PrimitiveLiteral` elements into `ParameterLookup` instances. This is not an issue with `MapProjectionLiteral` because it happens to such elements in other classes like `MapLiteral` as well.